src/passes/gccrs_rustc_successes.rs

```diff
@@ -0,0 +1,104 @@
+use crate::args::Args;
+use crate::copy_rs_files;
+use crate::error::Error;
+use crate::passes::{Pass, TestCase};
+
+use std::fmt::{Display, Formatter, Result as FmtResult};
+use std::fs;
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
+use std::time::Duration;
+
+use wait_timeout::ChildExt;
+
+pub enum GccrsRustcSuccesses {
+    Full,
+    NoStd,
+    NoCore,
+}
+
+impl Display for GccrsRustcSuccesses {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
+        let s = match self {
+            GccrsRustcSuccesses::Full => "",
+            GccrsRustcSuccesses::NoStd => "no-std",
+            GccrsRustcSuccesses::NoCore => "no-core",
+        };
+
+        write!(f, "{}", s)
+    }
+}
+
+impl Pass for GccrsRustcSuccesses {
+    fn fetch(&self, args: &Args) -> Result<Vec<PathBuf>, Error> {
+        // FIXME: Do we want more tests than just src/test/ui?
+        let rust_path = &args.rust_path;
+        let ui_tests = rust_path.join("src").join("test").join("ui");
+
+        let out_dir = match self {
+            GccrsRustcSuccesses::Full => args.output_dir.clone(),
+            // We need different output files since we're going to modify files for these
+            GccrsRustcSuccesses::NoStd => args.output_dir.join("no-std"),
+            GccrsRustcSuccesses::NoCore => args.output_dir.join("no-core"),
+        };
+
+        copy_rs_files(&ui_tests, &out_dir, rust_path)
+    }
+
+    fn adapt(&self, args: &Args, file: &Path) -> Result<TestCase, Error> {
+        let test_content = fs::read_to_string(file)?;
+
+        // FIXME: We should also see if the file contains a main function maybe?
+        // To know if we can pass `--crate-type=lib`?
+
+        // We're only interested in successes
+        if test_content.contains("ERROR") {
+            return Ok(TestCase::default());
```

Comment:
IIUC you return an empty ``TestCase`` as a kind of a error state. Why not add another ``Error`` variant and return that?

Reply:
The idea behind returning an "empty" test case and then checking for it afterwards (in your next comment) is that it is still valid - we simply want to skip this file and not generate a proper assertion for it. `try_fold` short circuits the execution on `Err` values, so we would end up not generating all the test cases we want. Unless I'm misunderstanding something about the way `try_fold` works, which is entirely possible :D

Reply:
I agree that the naming and checking afterwards could be better. I'll work on something!